### PR TITLE
fix(console): catch user interaction errors

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

catch user interaction errors
Note: in the application, all HTTP errors, except for 'loadTenants', are caused by user actions.

## Testing

image

image

## Checklist

- [ ] `.changset-staged`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

**Review by charIeszhao — APPROVED** [review]
LGTM. But better wait for another approver, making sure everyone is on the same page

File: packages/console/src/cloud/pages/Main/index.tsx [hunk 1]
```diff
@@ -18,8 +18,13 @@ function Protected() {
...
   useEffect(() => {
     const loadTenants = async () => {
-      const data = await api.get('/api/tenants').json<TenantInfo[]>();
-      setTenants(data);
+      try {
+        const data = await api.get('/api/tenants').json<TenantInfo[]>();
+        setTenants(data);
+      } catch {
+        // Note: this is intended to cause the app to crash.
+        throw new Error('Failed to load tenants.');
```
Comment (wangsijie) [inline packages/console/src/cloud/pages/Main/index.tsx:26]: The original error won't work?

Comment (xiaoyijun) [reply packages/console/src/cloud/pages/Main/index.tsx:26]: The original error may be an HTTP error, and we should crash the app in this case.
Other HTTP errors come from user interactions.

File: packages/console/src/cloud/pages/Main/index.tsx [hunk 2]
```diff
@@ -18,8 +19,16 @@ function Protected() {
...
-      const data = await api.get('/api/tenants').json<TenantInfo[]>();
-      setTenants(data);
+      try {
+        const data = await api.get('/api/tenants').json<TenantInfo[]>();
+        setTenants(data);
+      } catch (error: unknown) {
+        if (error instanceof HTTPError && error.response.status === 401) {
+          throw error;
+        }
+        // Note: this is intended to cause the app to crash.
```
Comment (charIeszhao) [inline packages/console/src/cloud/pages/Main/index.tsx:29]: We need to rephrase the comment. It's not for the app "to crash", but throw a non-http error instead in order to show the error details in the ErrorBoundary.

Comment (xiaoyijun) [reply packages/console/src/cloud/pages/Main/index.tsx:29]: Removed for we have a new implementation.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**